src/main/scala/com/rbmhtechnology/calliope/EventDeletionSink.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2015 - 2017 Red Bull Media House GmbH <http://www.redbullmediahouse.com> - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.calliope
+
+import akka.Done
+import akka.stream.scaladsl.Sink
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler}
+import akka.stream.{Attributes, Inlet, SinkShape}
+
+import scala.concurrent.{Future, Promise}
+import scala.util.{Failure, Success, Try}
+
+/**
+  * An EventDeletionSink is an Akka Streams Sink which deletes events through an instance of [[EventDeleter]].
+  * The Sink accepts elements which implement [[Sequenced]] and passes the sequence number of the element to the deletion request sent to the [[EventDeleter]].
+  *
+  * The Sink may be materialized to a [[Future]] to signal completion.
+  */
+object EventDeletionSink {
+  def apply[A: Sequenced](eventWriter: EventDeleter): Sink[A, Future[Done]] =
+    Sink.fromGraph(new EventDeletionSink(eventWriter))
+}
+
+final class EventDeletionSink[In: Sequenced](eventWriter: EventDeleter)(implicit sequenced: Sequenced[In]) extends GraphStageWithMaterializedValue[SinkShape[In], Future[Done]] {
```

Comment:
1) Very good catch - I think this should be doable without a problem. We could then extend the EventDeletion to offer a flow or a sink - something like this:

```
object EventDeletion {
  def flow[A: Sequenced](eventDeleter: EventDeleter)(implicit sequenced: Sequenced[A]): Flow[A, A, NotUsed]
  def sink[A: Sequenced](eventDeleter: EventDeleter)(implicit sequenced: Sequenced[A]): Sink[A, Future[Done]] 
}
```

2) I'd propose to perform the batching before applying this sink to be more flexible. We could provide a `SequencedBatch` though, that can be used for batching items of `Sequenced`. This batch could then be supplied to the sink.